src/controllers/machine/index.ts

```diff
@@ -34,9 +34,12 @@ export default class MachineController {
     }
 
     try {
-      const category = await CategoriesService.find({ id: body.categories[0] });
+      const categories = await Promise.all(
+        body.categories.map(async (categoryId) => CategoriesService.find({ id: categoryId }))
```

Comment:
what if body.categories is not sent on the request?

`body.categories?.map(...) || []`